ksqldb-common/src/main/java/io/confluent/ksql/config/KsqlConfigResolver.java

```diff
@@ -132,11 +133,21 @@ private static Optional<ConfigItem> resolveConfig(
 
     final String keyNoPrefix = stripPrefix(propertyName, prefix);
     final ConfigKey configKey = def.configKeys().get(keyNoPrefix);
-    if (configKey == null) {
-      return Optional.empty();
+    if (configKey != null) {
+      return Optional.of(ConfigItem.resolved(configKey));
+    }
+
+    if (isInternalStreamsConfig(keyNoPrefix)) {
+      return Optional.of(ConfigItem.unresolved(keyNoPrefix));
     }
 
-    return Optional.of(ConfigItem.resolved(configKey));
+    return Optional.empty();
+  }
+
+  private static boolean isInternalStreamsConfig(final String key) {
+    // add more internal configs on demand
+    return key.equals(InternalConfig.EMIT_INTERVAL_MS_KSTREAMS_OUTER_JOIN_SPURIOUS_RESULTS_FIX)
+        || key.equals(InternalConfig.EMIT_INTERVAL_MS_KSTREAMS_WINDOWED_AGGREGATION);
```

Comment:
Need this one to run QTTs...

Reply:
Is this only meant for QTT or are we expecting users to want/be able to set this for their own queries too? (Guessing the latter but want to check.)

Also, the first config (outer join spurious results fix emit interval) is unrelated to this PR, right? Is that another config we think users might want to set?

Reply:
Both are internal KS configs -- they control how often we try to emit results -- the other configs is for left/outer stream-stream joins (yes, it's unrelated to this PR, but serves the exact same purpose for a different operator).

Because both are internal, we don't really expect users to set them (with the exception that there is a perf problem and we tell them to change it). But we should be able for set the configs IMHO -- otherwise, the guard they provide for KS does not work for ksqlDB.

There are a few more internal configs, and we could consider to add them here, too.

Reply:
That makes sense. Just to confirm: users _will_ be able to set these internal streams configs on their ksql queries (and have the configs take effect) after this change, right? 

I think it's fine to leave the second config in -- we're essentially saying it's safe for users to set these configs on their ksql queries if they want to.

Reply:
Technically, yes.

Of course, both configs are "internal" meaning it's not documented anywhere (on purpose) and user most likely don't know about them (and should not need to know about them in general).